ngageoint/opensphere
# feat(storage): add persistent storage permission check

Browsers have internal limits for storage that may be cleared when under storage pressure. However, a recently-added permission allows the storage to be permanent.

This currently asks for the permission on every load (which might be overly aggressive).

See https://developers.google.com/web/updates/2016/06/persistent-storage for details.

## Testing
1. Serve from `http://` (not localhost) and verify that there are messages in the logs stating that persistent storage is not permitted in an unsecure environment
1. Firefox: Serve from `https://` (not localhost) and verify that the user is prompted for a persistent storage permission
    * Verify that declining the permission produces a warning in both the toast alerts and in the log
    * Verify that accepting the permission produces a successful log message and that `navigator.storage.persist()` resolves to `true` afterward
1. Chrome: Serve from `https` on a hostname that you don't hit often
    * Verify that it asks for notification permissions, and that accepting the permissions results in `navigator.storage.persist()` resolving to `true`
